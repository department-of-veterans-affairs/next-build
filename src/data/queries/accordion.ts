import { QueryFormatter } from 'next-drupal-query'
import { ParagraphAccordion } from '@/types/drupal/paragraph'
import { Accordion } from '@/types/formatted/accordion'

export const formatter: QueryFormatter<ParagraphAccordion, Accordion> = (
  entity: ParagraphAccordion
) => {
  return {
    id: entity.id,
    header: entity.field_header || null,
    html: entity.field_rich_wysiwyg.processed,
  }
}
