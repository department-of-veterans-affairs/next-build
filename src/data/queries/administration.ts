import { FieldAdministration } from '@/types/drupal/field_type'

export interface Administration {
  entityId: number
  name: string | null
}

/**
 * Formats a Drupal administration field into our standardized Administration interface.
 * 
 * In our Drupal content, the `field_administration` field points to a taxonomy Term
 * that is currently called "Section", but "Section" is only a human label added to
 * it in the Drupal interface, and that name has evolved over time. It was originally,
 * "administration" (and thus the name of the field). Since the only machine name of
 * it is `field_administration`, we'll continue to call it "administration" in our code.
 */
export const formatAdministration = (
  administration: FieldAdministration | null | undefined
): Administration => ({
  entityId: administration?.drupal_internal__tid || null, // "t" is for taxonomy term
  name: administration?.name || null,
})
