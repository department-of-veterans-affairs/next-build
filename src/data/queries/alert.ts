// Define the query params for fetching block--alert.
import { BlockAlert } from '@/types/drupal/block'
import { QueryFormatter } from 'next-drupal-query'
import { Alert } from '@/types/formatted/alert'

export const formatter: QueryFormatter<BlockAlert, Alert> = (
  entity: BlockAlert
) => {
  return {
    alertType:
      entity.field_alert_type === 'information'
        ? 'info'
        : entity.field_alert_type,
    id: entity.id,
    title: entity.field_alert_title,
    content: {
      header: entity.field_alert_content.field_text_expander,
      text: entity.field_alert_content.field_wysiwyg.processed,
    },
  }
}
