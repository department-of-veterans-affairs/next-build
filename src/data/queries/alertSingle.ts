import { QueryFormatter } from 'next-drupal-query'
import { ParagraphAlertSingle } from '@/types/drupal/paragraph'
import {
  AlertSingle,
  AlertBlock,
  AlertNonReusable,
} from '@/types/formatted/alert'
import { formatParagraph } from '@/lib/drupal/paragraphs'
import { queries } from '.'

// TODO:
//
// EDIT:
// It seems *maybe* that next-drupal-query is designed to handle this already.
// You can pass an `id` to queries.getParams().
// E.g.
// queries.getParams('paragraph--alert_single'))
//
// We don't need this here:
// export const params: QueryParams<null> = () => {
//   return queries
//     .getParams()
//     .addInclude([
//       'field_alert_block_reference',
//       'field_alert_block_reference.field_alert_content',
//       'field_alert_non_reusable_ref',
//       'field_alert_non_reusable_ref.field_va_paragraphs',
//     ])
// }
//
// ...but it seems like we might want to define
// paragraph-specific `includes` within the paragraph's
// query file (this file), and then somehow reference these in
// the node query when we need to include fields of a specific
// paragraph type.
//
// E.g.
// export const include = [
//   'field_alert_block_reference',
//   'field_alert_block_reference.field_alert_content',
//   'field_alert_non_reusable_ref',
//   'field_alert_non_reusable_ref.field_va_paragraphs',
// ]
//
// /* Some util file*/
// export const getIncludedSubFields = (
//   parentField: string,
//   subFields: string[]
// ) => [parentField, ...subFields.map((subField) => `${parentField}.${subField}`)]
//
// /* src/data/queries/resourcesSupport.ts */
// import { include } from '@/data/queries/alertSingle'
// export const params: QueryParams<null> = () => {
//   return queries.getParams().addInclude([
//     ...getIncludedSubFields('field_alert_single', include)
//   ])
// }

export const formatter: QueryFormatter<ParagraphAlertSingle, AlertSingle> = (
  entity: ParagraphAlertSingle
) => {
  return {
    type: entity.type as AlertSingle['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    alertSelection:
      entity.field_alert_selection as AlertSingle['alertSelection'],
    blockReference: queries.formatData(
      'block--alert',
      entity.field_alert_block_reference
    ) as AlertBlock,
    nonReusableRef: formatParagraph(
      entity.field_alert_non_reusable_ref
    ) as AlertNonReusable,
  }
}
