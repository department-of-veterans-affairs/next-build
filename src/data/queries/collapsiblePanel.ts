import { ParagraphCollapsiblePanel } from '@/types/drupal/paragraph'
import { QueryFormatter } from 'next-drupal-query'
import { CollapsiblePanel } from '@/types/formatted/collapsiblePanel'
import { formatParagraph } from '@/lib/drupal/paragraphs'

export const formatter: QueryFormatter<
  ParagraphCollapsiblePanel,
  CollapsiblePanel
> = (entity: ParagraphCollapsiblePanel) => {
  return {
    type: entity.type as CollapsiblePanel['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    paragraphs: entity.field_va_paragraphs?.map?.(
      formatParagraph<'paragraph--collapsible_panel_item'>
    ),
    bordered: entity.field_collapsible_panel_bordered,
    startExpanded: entity.field_collapsible_panel_expand,
    multiSelect: entity.field_collapsible_panel_multi,
  }
}
