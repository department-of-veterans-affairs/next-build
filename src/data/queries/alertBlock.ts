// Define the query params for fetching block--alert.
import { BlockAlert } from '@/types/drupal/block'
import { QueryFormatter } from 'next-drupal-query'
import { AlertBlock, AlertType } from '@/types/formatted/alert'

export const formatter: QueryFormatter<BlockAlert, AlertBlock> = (
  entity: BlockAlert
) => {
  if (!entity) {
    return null
  }

  return {
    alertType: (entity.field_alert_type === 'information'
      ? 'info'
      : entity.field_alert_type) as AlertType,
    id: entity.id,
    title: entity.field_alert_title,
    content: {
      header:
        entity.field_alert_content.field_text_expander ??
        entity.field_alert_heading,
      text: entity.field_alert_content?.field_wysiwyg?.processed ?? '',
    },
  }
}
