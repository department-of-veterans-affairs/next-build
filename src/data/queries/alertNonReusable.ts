// Define the query params for fetching block--alert.
import { BlockAlert } from '@/types/drupal/block'
import { QueryFormatter } from 'next-drupal-query'
import { AlertType, AlertNonReusable } from '@/types/formatted/alert'
import { ParagraphNonReusableAlert } from '@/types/drupal/paragraph'
import { formatParagraph } from '@/lib/drupal/paragraphs'

export const formatter: QueryFormatter<
  ParagraphNonReusableAlert,
  AlertNonReusable
> = (entity: ParagraphNonReusableAlert) => {
  if (!entity) {
    return null
  }

  return {
    type: entity.type as AlertNonReusable['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    alertType: (entity.field_alert_type === 'information'
      ? 'info'
      : entity.field_alert_type) as AlertType,
    heading: entity.field_alert_heading,
    paragraphs: entity.field_va_paragraphs.map(formatParagraph),
  }
}
