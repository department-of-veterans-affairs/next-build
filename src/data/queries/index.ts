import { createQueries } from 'next-drupal-query'
import * as NewsStory from './newsStory'
import * as NewsStoryTeaser from './newsStoryTeaser'
import * as StoryListing from './storyListing'
import * as QuestionAnswer from './questionAnswer'
import * as ExpandableText from './expandableText'
import * as LinkTeaser from './linkTeaser'
import * as MediaImage from './mediaImage'
import * as Banner from './banners'
import * as PersonProfile from './personProfile'
import * as Button from './button'
import * as AudienceTopics from './audienceTopics'
import * as Alert from './alert'
import * as EmailContact from './emailContact'
import * as BenefitsHub from './benefitsHub'
import * as Wysiwyg from './wysiwyg'
import * as StaticPathResources from './staticPathResources'
import * as HeaderFooter from './headerFooter'

export const queries = createQueries({
  // standard Drupal entity data queries
  'node--news_story': NewsStory,
  'node--news_story--teaser': NewsStoryTeaser,
  'node--story_listing': StoryListing,
  'node--q_a': QuestionAnswer,
  'node--person_profile': PersonProfile,
  'node--landing_page': BenefitsHub, // "Benefits Hub Landing Page"
  'paragraph--audience_topics': AudienceTopics,
  'paragraph--button': Button,
  'paragraph--email_contact': EmailContact,
  'paragraph--expandable_text': ExpandableText,
  'paragraph--link_teaser': LinkTeaser,
  'paragraph--rich_text_char_limit_1000': Wysiwyg,
  'paragraph--wysiwyg': Wysiwyg,
  'media--image': MediaImage,
  'block--alert': Alert,

  // Custom queries 
  'banner--alerts_lookup': Banner,
  'header-footer-data': HeaderFooter,

  // Static Path Generation
  'static-path-resources': StaticPathResources,
})
