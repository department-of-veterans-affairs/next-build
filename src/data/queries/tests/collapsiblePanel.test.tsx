/**
 * @jest-environment node
 */

import { ParagraphCollapsiblePanel } from '@/types/drupal/paragraph'
import { queries } from '@/data/queries'
import { mockResponse } from '@/mocks/collapsiblePanel.mock'
import { params } from '../collapsiblePanel'
const CollapsiblePanel: ParagraphCollapsiblePanel = mockResponse

describe('DrupalJsonApiParams configuration', () => {
  test('params function sets the correct include fields', () => {
    const paramsInstance = params()
    const queryString = decodeURIComponent(paramsInstance.getQueryString())
    expect(queryString).toMatch(/include=field_va_paragraphs/)
  })
})

describe('Collapsible Panel formatData', () => {
  test('outputs formatted data', () => {
    expect(
      queries.formatData('paragraph--collapsible_panel', CollapsiblePanel)
    ).toMatchSnapshot()
  })
})
