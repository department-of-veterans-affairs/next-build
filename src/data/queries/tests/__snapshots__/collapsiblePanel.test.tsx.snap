// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsible Panel formatData outputs formatted data 1`] = `
{
  "bordered": false,
  "entityId": 2862,
  "id": "cc597ad6-123b-49b4-bf90-2edae8b355bc",
  "multiSelect": false,
  "paragraphs": [
    {
      "entityId": 2861,
      "id": "768a4d33-0552-49d8-a9e1-c66d5ae689f5",
      "paragraphs": [],
      "title": "Homeless services",
      "type": "paragraph--collapsible_panel_item",
      "wysiwyg": "<h2>24/7 VA helpline for homeless or at-risk Veterans: 877-4AID-VET (877-424-3838)</h2>

<p><strong>@Shelley's team&nbsp;</strong>If functionality is available, this could be an alert box or similar component</p>

<p>If you are a Veteran who is homeless or at risk of becoming homeless due to financial hardship, unemployment, addiction, depression, or transition from jail, VA Pittsburgh can help you with:</p>

<ul>
	<li>Immediate food and shelter needs</li>
	<li>Transitional and permanent housing</li>
	<li>Job training and life skills development</li>
	<li>Educational opportunities</li>
	<li>Support with justice system navigation</li>
	<li>Re-entry from jail into the community</li>
	<li>Financial support to prevent homelessness</li>
	<li>Treatment for addiction and depression</li>
	<li>Health and dental care</li>
	<li>Case management</li>
</ul>

<p><strong>@Shelley's team</strong> may want to consider whether different care coordinators should be listed on different accordions. For example, should Mary Frances be listed here because she handles all general incoming requests for homeless services, while Linda should be listed on the "permanent housing" accordion because she handles HUD-VASH cases?</p>

<p>For local help getting started with any of these services, contact a VA Pittsburgh homeless services care coordinator. We can help determine what programs are right for you.</p>

<h3>Homeless services care coordinators</h3>

<p>Mary Frances Pilarski<br>
Coordinator for homeless Veterans health care<br>
412-822-1272 (direct)<br>
866-482-7488, ext. 221272 (outside the Pittsburgh area)<br>
<a href="mailto:Mary.Pilarski@va.gov">Mary.Pilarski@va.gov</a></p>

<p>Keather Likins<br>
Outreach social worker<br>
412-822-1409<br>
<a href="mailto:Keather.Likins@va.gov">Keather.Likins@va.gov</a></p>

<p>Linda Crawford<br>
HUD-VASH social worker<br>
412-822-1271<br>
<a href="mailto:linda.crawford2@va.gov">linda.crawford2@va.gov</a></p>

<p>Joseph Savino<br>
Transitional housing liaison<br>
412-822-1295<br>
<a href="mailto:Joseph.Savino2@va.gov">Joseph.Savino2@va.gov</a></p>

<p>Mary Henderson<br>
Veterans benefits counselor<br>
412-822-1275<br>
&lt;insert email&gt;</p>

<p>You can also come in person to this office at this location to talk to someone about which homeless services are right for you.</p>

<h3>For Veterans struggling with addiction or substance abuse</h3>

<p>You can enroll in our intensive 14 or 21-day behavioral health program to start your recovery from alcohol or drug addiction. Call us at 412-360-6110 or 412-360-6092</p>

<h3>Center for Treatment of Addictive Disorders</h3>

<p>University Drive campus [Link to: University Drive campus page, API]<br>
First floor, check-in area 3</p>

<h3>For help with finding and keeping a job</h3>

<p>We can help you develop the skills to find and keep a job. We can also connect you with VA programs that allow you to earn a competitive wage (such as the state minimum wage).</p>

<h4>Contact</h4>

<hr>
<p><strong>@Shelley team:&nbsp;</strong>For contacts like this, we need to answer these questions: Does Joseph only handle transitional housing cases? (If so, only list him with the transitional housing info.) Does John act as a care coordinator for ALL homeless services? (If so, consider listing him here but leaving off the title "Transitional housing liaison", which is distracting to the user and makes them wonder if he's the right person to be contacting.)</p>

<p><strong>@Shelley team:</strong> This name and number is on a PDF brochure for Pittsburgh VA homeless services, but not on the web page. Not sure whether to include.</p>

<p><strong>@Shelley team:</strong> to fill in, if this is a viable way to get help with homeless services</p>

<p><strong>@Shelley team:&nbsp;</strong>Both these numbers are listed in the directory. Only the first is listed on the website page. Which is better? Why?</p>

<p><strong>@Shelley team:&nbsp;</strong>Giving the name and location of this program here might be distracting, as the CTA is really just to call the number.</p>

<p>John Erskine<br>
412-822-1285<br>
<a href="mailto:John.Erskine@va.gov">John.Erskine@va.gov</a></p>

<h3>For Veterans with justice involvement or transitioning out of prison</h3>

<p>We connect Veterans in local courts and jails with services to help them reintegrate back into their communities as smoothly as possible. We can provide housing and employment opportunities, health care, and access to VA benefits. Call us at 412-822-1275.</p>

<h3>For Veterans with psychiatric disorders</h3>

<p>We provide residential programs for Veterans struggling with psychiatric disorders and help them transition back into their communities. Contact:</p>

<p>Bridget Wright<br>
412-822-1403<br>
&lt;email&gt;</p>

<p><a>Tim Morrison<br>
412-822-1399</a><br>
<a href="mailto:Timothy.Morrison@va.gov">Timothy.Morrison@va.gov</a></p>

<p><strong>@Shelley: </strong>Is this a common term? If I were a user I don’t know that I’d understand it.</p>

<p>Listed as contact on a resources page for homeless Veterans</p>

<p>Listed on brochure for the Recovery Center for Veterans under the header for Residential Treatment for Psychiatric Disorders</p>

<p><strong>@Shelley: </strong>Can we specify what they do/what their titles are?</p>",
    },
  ],
  "startExpanded": false,
  "type": "paragraph--collapsible_panel",
}
`;
